spec/lib/miq_automation_engine/models/miq_ae_domain_spec.rb

```diff
@@ -176,29 +176,49 @@
     it "import a domain" do
       domain_name = dom1.name
       expect(MiqAeImport).to receive(:new).with(any_args).and_return(git_import)
-      expect(repo).to receive(:branch_info).with(branch_name).and_return(info)
+      expect_any_instance_of(GitRepository).to receive(:branch_info).with(branch_name).and_return(info)
       allow(git_import).to receive(:import) { MiqAeDomain.create(:name => domain_name) }
+      options = {'domain' => domain_name, 'git_repository_id' => repo.id,
+                 'tenant_id' => @user.current_tenant.id, 'ref' => branch_name,
+                 'ref_type' => 'BrancH'}
+      dom1 = MiqAeDomain.import_git_repo(options)
+      expect(dom1.attributes).to have_attributes(commit_hash)
+    end
+
+    it "import domain embedded in git repository" do
+      domain_name = "ASTERIX"
+      expect(MiqAeImport).to receive(:new).with(any_args).and_return(git_import)
+      expect_any_instance_of(GitRepository).to receive(:branch_info).with(branch_name).and_return(info)
+      allow(git_import).to receive(:import) { [MiqAeDomain.create(:name => domain_name)] }
+      options = {'git_repository_id' => repo.id,
+                 'tenant_id' => @user.current_tenant.id, 'ref' => branch_name,
+                 'ref_type' => "BRANCH" }
+      dom1 = MiqAeDomain.import_git_repo(options)
 
-      MiqAeDomain.import_git_repo(domain_name, repo, branch_name)
-      dom1 = MiqAeDomain.find_by(:name => domain_name)
       expect(dom1.attributes).to have_attributes(commit_hash)
     end
 
     it "import a domain fails" do
-      domain_name = dom1.name
       expect(MiqAeImport).to receive(:new).with(any_args).and_return(git_import)
-      allow(git_import).to receive(:import) { MiqAeDomain.create(:name => "nada") }
+      allow(git_import).to receive(:import) { nil }
       expect do
-        MiqAeDomain.import_git_repo(domain_name, repo, branch_name)
+        options = {'git_repository_id' => repo.id,
+                   'tenant_id' => @user.current_tenant.id, 'ref' => branch_name}
+        MiqAeDomain.import_git_repo(options)
       end.to raise_error(MiqAeException::DomainNotFound)
     end
 
+    it "import without git_repository_id" do
+      expect { MiqAeDomain.import_git_repo({}) }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+
+
     it "git repo changed for non git domain" do
       expect { dom2.git_repo_changed? }.to raise_error(MiqAeException::InvalidDomain)
     end
 
     it "git repo branch changed" do
-      expect(repo).to receive(:branch_info).with(branch_name).twice.and_return(new_info)
+      expect_any_instance_of(GitRepository).to receive(:branch_info).with(branch_name).twice.and_return(new_info)
       dom1.update_attributes(:ref => branch_name, :git_repository => repo,
                              :ref_type => MiqAeDomain::BRANCH, :commit_sha => commit_sha)
       expect(dom1.git_repo_changed?).to be_truthy
```

Comment:
We probably need to talk as a team whether or not we want to go this direction, but personally I'd prefer if these tests were broken up via `context`s instead of simply `it`s. It may end up being more verbose, but it separates paths and shows you how you could (and/or should) refactor the code if your tests get too many levels deep.

That would make this file look something like this:
```
context "when importing from a git url" do
  # setup

  context "when the username/password is given" do
    # it blocks go here
  end

  context "when the username/password is not given" do
    # it blocks go here, might be the same as the above it blocks
  end
end

context "when the import domain is embedded in the git repository" do
  # setup

  it "finds the domain name and imports the domain" do
    # expectation
  end
end

context "when the git repository id doesn't exist" do
  # setup

  it "raises an error" do
    # expectation
  end
end
```

Like? Dislike? Indifferent?